packages/pnpm-sync-lib/src/pnpmSyncPrepare.ts

```diff
@@ -85,27 +87,29 @@ export async function pnpmSyncPrepareAsync(options: IPnpmSyncPrepareOptions): Pr
     throw Error('The input pnpm-lock.yaml path or the input .pnpm folder path is not correct!');
   }
 
-  const startTime = process.hrtime.bigint();
-
-  // read the pnpm-lock.yaml
-  const pnpmLockfile: ILockfile | undefined = await readPnpmLockfile(lockfilePath, {
-    ignoreIncompatible: true
-  });
+  const pnpmModulesYamlPath: string = path.resolve(dotPnpmFolder, '..');
+  const pnpmModulesYaml = YAML.parse(fs.readFileSync(`${pnpmModulesYamlPath}/.modules.yaml`, 'utf8'));
+  const pnpmVersion: string | undefined = pnpmModulesYaml?.packageManager?.split('@')[1];
 
-  // currently, only support lockfileVersion 6.x, which is pnpm v8
-  const lockfileVersion: string | undefined = pnpmLockfile?.lockfileVersion.toString();
```

Comment:
I think you still need to check `lockfileVersion` as well. 🙂

Even if it is true that PNPM 8 always uses lockfile version 6 (is this really always guaranteed?), it is easy to imagine ways that the lockfile might have been written by a different version of PNPM than `.modules.yaml`.  For example, `git checkout` might rewrite the lockfile arbitrarily, right?